File: Source/UnitTests/TestCode/AccessibilityPropagatorTransformerTest/PublicMethodInheritedProtected.java
package Test;
public abstract class Shape
{
	protected abstract void FillColor();
}
public abstract class SimpleShape extends Shape
{
}
public class Rectangle extends SimpleShape 
{
	public void FillColor()
	{
	}
}

public class PolygonShape extends Shape 
{
	protected void FillColor()
	{
	}
}
public class ComplexShape extends PolygonShape
{
	protected void FillColor()
	{
	}
}

File: Source/UnitTests/TestCode/AccessibilityPropagatorTransformerTest/PublicMethodInheritedProtected.cs
namespace Test
{
	public abstract class Shape
	{
		public abstract void FillColor();
	}
	public abstract class SimpleShape : Shape
	{
	}
	public class Rectangle : SimpleShape 
	{
		public void FillColor()
		{
		}
	}

	public class PolygonShape : Shape 
	{
		public void FillColor()
		{
		}
	}
	public class ComplexShape : PolygonShape
	{
		public void FillColor()
		{
		}
	}
}